### PR TITLE
Problem: empty topics in rpc response is not tested

### PR DESCRIPTION
**Pull request by yihuang**
backport: https://github.com/crypto-org-chain/cronos/pull/254

Solution:
- merge empty topics fix
- add integration tests to verify

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -164,4 +164,4 @@ replace github.com/ethereum/go-ethereum => github.com/crypto-org-chain/go-ethere
 // TODO: remove when ibc-go and ethermint upgrades cosmos-sdk
 replace github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.44.2
 
-replace github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.7.2-cronos-4
+replace github.com/tharsis/ethermint => github.com/yihuang/ethermint v0.6.1-0.20211214073201-9fe3fe860947
```
Comment (yihuang) [inline go.mod:167]: TODO, update after https://github.com/crypto-org-chain/ethermint/pull/35 merged and new tag created.

File: go.mod [hunk 2]
```diff
@@ -164,4 +164,4 @@ replace github.com/ethereum/go-ethereum => github.com/crypto-org-chain/go-ethere
 // TODO: remove when ibc-go and ethermint upgrades cosmos-sdk
 replace github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.44.2
 
-replace github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.7.2-cronos-4
+replace github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.7.2-cronos-5
```
Comment (thomas-nguy) [inline go.mod:167]: we can probably also add the ante handler fix at the same time

Comment (yihuang) [reply go.mod:167]: that one should be fixed privately first I think.

File: integration_tests/shell.nix [hunk 3]
```diff
@@ -3,7 +3,7 @@ pkgs.mkShell {
   buildInputs = [
     pkgs.jq
     pkgs.go
-    (import ../. { inherit pkgs; }) # cronosd
+    # (import ../. { inherit pkgs; }) # cronosd
```
Comment (yihuang) [inline integration_tests/shell.nix:6]:
```suggestion
    (import ../. { inherit pkgs; }) # cronosd
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,10 @@
 - [cronos#255](https://github.com/crypto-org-chain/cronos/pull/255) fix empty topics in non-breaking way
 - [cronos#270](https://github.com/crypto-org-chain/cronos/pull/270) reject MsgEthereumTx wrapping tx without the extension option.
 
+### Bug Fixes
+
+- [cronos#255](https://github.com/crypto-org-chain/cronos/pull/255) fix empty topics in non-breaking way
+
```
Comment (yihuang) [inline CHANGELOG.md:13]:
```suggestion
```

**Review by leejw51crypto — APPROVED** [review]
lgtm

**Review by greg9762 — APPROVED** [review]

**Pull request merged**